e2e_playwright/conftest.py

```diff
@@ -248,6 +248,32 @@ def app(page: Page, app_port: int) -> Page:
     return page
 
 
+@pytest.fixture(scope="function")
+def app_with_params(page: Page, app_port: int, request) -> Tuple[Page, Dict]:
+    """Fixture that opens the app with additional query parameters.
+    The query parameters are passed as a dictionary in the 'query_params' key of the request.
+    """
+    query_params = request.param
+
+    from urllib import parse
```

Comment:
import should be at the top of the method. Also, is this something that has to happen here, or can we move it to the top of the file?